util/devel/look_for_calls.py

```diff
@@ -122,18 +122,35 @@ def check_for_calls(functions, search_dir, exclude_paths=None, rel_paths=True):
 
         found_calls = False
         for func in functions:
-            out = cscope_find_calls(func, cscope_database_name)
-            if out:
-                found_calls = True
-                log_error('found call to "{0}"'.format(func))
-                sys.stdout.write(out.replace(rel_dir, '') + '\n')
+          # If func is a tuple consider the first element to be the function
+          # we're searching for and the second an alternative to suggest
+          # to the user.
+          alternative = None
+          if isinstance(func, tuple):
+            alternative = func[1]
+            func = func[0]
+            
+          out = cscope_find_calls(func, cscope_database_name)
+          if out:
+              found_calls = True
+              msg = 'found call to "{0}"'.format(func)
+              if alternative is not None:
+                msg += ' consider using "{0}" instead'.format(alternative)
+              log_error(msg)
+              sys.stdout.write(out.replace(rel_dir, '') + '\n')
 
         return found_calls
 
 
 def get_alloc_funcs():
     """Return a list of the possible C alloc/dealloc routines"""
-    std = ['malloc', 'calloc', 'realloc', 'free']
+    # If list element is a tuple then consider the second element to
+    # be an alternative for the first. If the list element is a
+    # not a tuple then their is no known alternative.
+    std = [('malloc', 'chpl_malloc'),
+           ('calloc', 'chpl_calloc'),
+           ('realloc', 'chpl_realloc'),
+           ('free', 'chpl_free')]
```

Comment:
`chpl_malloc` and friends is probably not right here. I think instead we'd want `chpl_mem_alloc` and friends. The difference being that these `mem_` versions are what do memory tracking (look for leaks, etc.)

@gbtitus Does that sound right to you?

Reply:
Yes, `chpl_malloc()` etc. are direct calls to the runtime memory layer implementations, bypassing the tracking support. But in almost all circumstances we actually want memory that can be tracked if the user enabled that, so `chpl_mem_alloc()` etc. are the desired interface.

Reply:
Thanks, I've updated the suggestions to use `chpl_mem_alloc` and friends.

Reply:
Can you update the PR description too?

Reply:
Done.